xbmc/utils/ScraperParser.cpp

```diff
@@ -285,7 +285,7 @@ void CScraperParser::ParseExpression(const std::string& input, std::string& dest
 
       if (iOptional > -1) // check that required param is there
       {
-        char temp[4];
+        char temp[12];
```

Comment:
This is more than a minor :-)

Reply:
I know, but wasn't in the mood of creating a separate PR for the two string fixes.
All fine if I keep it in this PR?

Reply:
Perfectly fine :-)

Reply:
snprintf would also be enough. optional is never > 2 digits, obviously it can be evil crafted to be so but ;p